File: backend/app/routes/comments.py
from flask import Blueprint, request, jsonify
from flask_jwt_extended import jwt_required, get_jwt_identity
from app.services.comment_service import CommentService

bp = Blueprint('comments', __name__)


@bp.route('/tasks/<int:task_id>/comments', methods=['GET'])
@jwt_required()
def get_task_comments(task_id):
    """Получить все комментарии задачи"""
    current_user_id = get_jwt_identity()
    
    comments, error = CommentService.get_task_comments(task_id, current_user_id)
    
    if error:
        return jsonify({'error': error}), 404
    
    return jsonify({
        'comments': [c.to_dict() for c in comments]
    }), 200


@bp.route('/tasks/<int:task_id>/comments', methods=['POST'])
@jwt_required()
def create_comment(task_id):
    """Создать комментарий к задаче"""
    current_user_id = get_jwt_identity()
    data = request.get_json()
    
    if not data:
        return jsonify({'error': 'No data provided'}), 400
    
    comment, error = CommentService.create_comment(task_id, data, current_user_id)
    
    if error:
        return jsonify({'error': error}), 400
    
    return jsonify({
        'message': 'Comment created successfully',
        'comment': comment.to_dict()
    }), 201


@bp.route('/comments/<int:comment_id>', methods=['PUT'])
@jwt_required()
def update_comment(comment_id):
    """Обновить комментарий"""
    current_user_id = get_jwt_identity()
    data = request.get_json()
    
    if not data:
        return jsonify({'error': 'No data provided'}), 400
    
    comment, error = CommentService.update_comment(comment_id, data, current_user_id)
    
    if error:
        return jsonify({'error': error}), 400
    
    return jsonify({
        'message': 'Comment updated successfully',
        'comment': comment.to_dict()
    }), 200


@bp.route('/comments/<int:comment_id>', methods=['DELETE'])
@jwt_required()
def delete_comment(comment_id):
    """Удалить комментарий"""
    current_user_id = get_jwt_identity()
    
    success, error = CommentService.delete_comment(comment_id, current_user_id)
    
    if error:
        return jsonify({'error': error}), 400
    
    return jsonify({
        'message': 'Comment deleted successfully'
    }), 200

File: frontend/src/api/comments.ts
import { api } from "./axios";
import { Comment } from "@/types";

export const commentsApi = {
  // Создать комментарий
  createComment: async (taskId: number, content: string, parentId?: number): Promise<Comment> => {
    const response = await api.post(`/comments/task/${taskId}`, {
      content,
      parent_id: parentId,
    });
    return response.data.comment || response.data;
  },

  // Получить комментарии задачи
  getTaskComments: async (taskId: number): Promise<Comment[]> => {
    const response = await api.get(`/comments/task/${taskId}`);
    return Array.isArray(response.data) ? response.data : response.data.comments || [];
  },

  // Редактировать комментарий
  updateComment: async (commentId: number, content: string): Promise<Comment> => {
    const response = await api.put(`/comments/${commentId}`, { content });
    return response.data.comment || response.data;
  },

  // Удалить комментарий
  deleteComment: async (commentId: number): Promise<void> => {
    await api.delete(`/comments/${commentId}`);
  },

  // Получить ответы на комментарий
  getCommentReplies: async (commentId: number): Promise<Comment[]> => {
    const response = await api.get(`/comments/${commentId}/replies`);
    return Array.isArray(response.data) ? response.data : response.data.replies || [];
  },
};
